spotbugs/src/main/java/edu/umd/cs/findbugs/util/ClassName.java

```diff
@@ -175,9 +175,47 @@ public static String extractSimpleName(@DottedClassName String className) {
      * @return true if it's a valid class name, false otherwise
      */
     public static boolean isValidClassName(String className) {
-        // FIXME: should use a regex
+        return isValidBinaryClassName(className) ||
+                isValidDottedClassName(className) ||
+                isValidArrayFieldDescriptor(className) ||
+                isValidClassFieldDescriptor(className);
+    }
+
+    private static boolean isValidBinaryClassName(String className) {
```

Comment:
Better to have a hyperlink referring to the definition of 'valid binary class name':

```suggestion
    /**
     * @see <a href="https://docs.oracle.com/javase/specs/jvms/se8/html/jvms-4.html#jvms-4.2.1">JVMS (Java SE 8 Edition) 4.2.2. Unqualified Names</a>
     */
    private static boolean isValidBinaryClassName(String className) {
```

Note that we may need to run `./gradlew spotlessApply` to reformat my snippet.

Reply:
In your suggestion, the link is to section 4.2.1, but the link text is to 4.2.2. I think 4.2.1 is most correct, as it refers to class names, while 4.2.2 refers to the elements of the class name separated by `/` or `.`.

I also tried to wrap the link, as it resulted in a very long line.

Feel free to throw this back to me if you feel that 4.2.1 is incorrect as we should stick with 4.2.2 instead.

Reply:
Thanks for your confirmation! The following sentence in the "4.2.1. Binary Class and Interface Names" is correct reference:

> The identifiers themselves must be unqualified names (§4.2.2).